sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -237,7 +237,7 @@ class Dataset[T] private[sql](
    */
   private[sql] def showString(
       _numRows: Int, truncate: Int = 20, vertical: Boolean = false): String = {
-    val numRows = _numRows.max(0)
+    val numRows = _numRows.max(0).min(Int.MaxValue - 1)
```

Comment:
OK, but now you return one fewer row than expected when it's possible to return Int.MaxValue. Granted this is an extreme corner case, but that seems less compelling than just skipping the display of "more elements" in this case.

Reply:
hmm, I see. Both is okay to me and WDYT? cc: @gatorsmile 
IMHO it might be still okay to set `[0, Int.MaxValue)` as valid range for `show` cuz this is a super extreme corner case.

Reply:
Spark SQL does not work when the number of rows is close to `Int.MaxValue`. The driver will be OOM before finishing the command. Thus, I do not think we can hit this extreme case.